aws-datastore/src/main/java/com/amplifyframework/datastore/storage/sqlite/SQLiteStorageAdapter.java

```diff
@@ -642,44 +628,128 @@ private CreateSqlCommands getCreateCommands(@NonNull Set<String> modelNames) {
         return new CreateSqlCommands(createTableCommands, createIndexCommands);
     }
 
-    // Binds each value inside list onto compiled statement in order
-    private void bindStatementToValues(
-            @NonNull SqlCommand sqlCommand,
-            @Nullable Model model
-    ) throws DataStoreException {
-        final SQLiteStatement compiledSqlStatement = sqlCommand.getCompiledSqlStatement();
-        // 1-based index for columns
-        int columnIndex = 1;
+    // extract model field values to save in database
+    private List<Object> extractFieldValues(@NonNull Model model) throws DataStoreException {
+        final String modelName = getModelName(model);
+        final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
+        final SQLiteTable table = SQLiteTable.fromSchema(schema);
+        final SQLiteModelFieldTypeConverter converter =
+                new SQLiteModelFieldTypeConverter(schema, modelSchemaRegistry, gson);
+        final Map<String, ModelField> modelFields = schema.getFields();
+        final List<Object> bindings = new ArrayList<>();
+        for (SQLiteColumn column : table.getSortedColumns()) {
+            final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
+            final Object fieldValue = converter.convertValueFromTarget(model, modelField);
+            bindings.add(fieldValue);
+        }
+        return bindings;
+    }
 
-        // bind model field values to sql columns
-        if (model != null) {
-            final String modelName = getModelName(model);
-            final ModelSchema schema = modelSchemaRegistry.getModelSchemaForModelClass(modelName);
-            final SQLiteModelFieldTypeConverter converter =
-                    new SQLiteModelFieldTypeConverter(schema, modelSchemaRegistry, gson);
-            final Map<String, ModelField> modelFields = schema.getFields();
-
-            final List<SQLiteColumn> columns = sqlCommand.getColumns();
-
-            for (SQLiteColumn column : columns) {
-                final ModelField modelField = Objects.requireNonNull(modelFields.get(column.getFieldName()));
-                final Object fieldValue = converter.convertValueFromTarget(model, modelField);
-                bindValueToStatement(compiledSqlStatement, columnIndex, fieldValue);
-                columnIndex++;
+    private List<Model> cascade(Collection<? extends Model> models) {
```

Comment:
Minor naming suggestion?
```suggestion
    private List<Model> descendants(Collection<? extends Model> models) {
```

Or, does it make sense to have this method at all in the `SQLiteStorageAdapter`?  It is basically converting a `Map.Entry<ModelSchema, Set<String>>` to a `List<Model>`.  Would it be better to move this to the `SQLiteModelTree` utility class?  Maybe `descendantsOf` should just return a `List<Model>`?